cmd/ui/src/components/GraphEvents.tsx

```diff
@@ -76,11 +79,15 @@ export const GraphEvents: FC<GraphEventProps> = ({
                 setHoveredNode(null);
             },
             downNode: (event) => {
-                setDraggedNode(event.node);
-
-                dragTimerRef.current = setTimeout(() => {
-                    isLongPress.current = true;
-                }, 150);
+                // a button value of `0` indicates that the main button of the mouse device was pressed
+                // https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button#value
+                if (event.event.original.button === 0) {
+                    setDraggedNode(event.node);
+
+                    dragTimerRef.current = setTimeout(() => {
+                        isLongPress.current = true;
+                    }, 150);
+                }
```

Comment:
It looks like react-sigma offers both `clickNode` and `rightClickNode` events - would those provide a cleaner solution?

Reply:
is this comment resolved?

we need to have the logic that is currently defined in downNode() in a separate, discrete step from the logic defined in clickNode().  this is because sigma fires downNode() before it fires clickNode(), so our dragging-related logic goes into downNode().  it is mandatory to include the `event.original.button === 0` check in the downNode() handler because we only want to drag nodes on left clicks, not right clicks.